### PR TITLE
Add runc container list to API and working sysbox checkpoint

### PR DESCRIPTION
**Pull request by bsmithai**
## Describe your changes
- Fixed get all containers rpc call
- sysbox c/r changes added for new architecture (checkpoint a process under systemd)
- Added handleRestoringExternalPidNamespaces
- Added specific pid c/r option to cli for debugging

## Issue ticket number 
[CED-355](https://linear.app/cedana/issue/CED-355/fix-runc-container-list) and [CED-372](https://linear.app/cedana/issue/CED-372/cr-a-pytorch-job)

## TODO
- [x] needs synced with main for clean merge
- [x] state json root is not correct in testing, just need to pull containerid from bundle path
- [x] verify restore is in the same state as it was before using state.json

## Checklist before requesting a review
- [x] Have I performed a self-review?
- [ ] Have I added regression or unit tests (where it makes sense) for my changes?
- [ ] Have I updated the README if changes have resulted in it being out of date?
- [x]  Have I checked to ensure my PR only introduces the changes it's intended to? E.g. No extraneous formatting changes.
- [x] Is this a breaking change? If yes, please describe areas affected and request reviews from developer stakeholders.

### CODE REVIEW

File: scripts/k8s/setup-host.sh [hunk 1]
```diff
@@ -50,9 +50,11 @@ git clone https://github.com/cedana/cedana.git
 echo "export IS_K8S=1" >> ~/.bashrc
 source ~/.bashrc
 
-cd cedana && go build -v
+cd cedana
+git fetch && git checkout feat/ced-355
```
Comment (nravic) [inline scripts/k8s/setup-host.sh:54]: must checkout main

File: scripts/k8s/setup-host.sh [hunk 2]
```diff
@@ -50,9 +50,11 @@ git clone https://github.com/cedana/cedana.git
...
+go build -v
 cp cedana /usr/local/bin/cedana
 
-cedana daemon start &
+./build-start-daemon.sh
```
Comment (nravic) [inline scripts/k8s/setup-host.sh:58]: if you want to start this as systemctl, pass `--systemctl` as a flag

File: cedana-helper.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+RUN add-apt-repository ppa:criu/ppa
+RUN apt update && apt install -y criu python3 pip sudo iptables
+
+RUN git clone https://github.com/cedana/cedana && mkdir ~/.cedana
+WORKDIR /cedana
+
+RUN git fetch --all --tags && \
+    LATEST_TAG=$(git describe --tags `git rev-list --tags --max-count=1`) && \
+    git checkout feat/ced-355 && \
```
Comment (nravic) [inline cedana-helper.Dockerfile:20]: checking out a branch here

File: api/dump.go [hunk 4]
```diff
@@ -157,10 +158,12 @@ func (s *service) prepareDumpOpts() *rpc.CriuOpts {
 	return &opts
 }
 
-func (s *service) runcDump(ctx context.Context, root, containerID string, opts *container.CriuOpts, state *task.ProcessState) error {
+func (s *service) runcDump(ctx context.Context, root, containerID string, pid int32, opts *container.CriuOpts, state *task.ProcessState) error {
```
Comment (nravic) [inline api/dump.go:161]: does this break regular runc checkpointing? e.g not sysbox?

File: container/checkpoint.go [hunk 5]
```diff
@@ -1311,14 +1311,22 @@ func (c *RuncContainer) RuncCheckpoint(criuOpts *CriuOpts, pid int, runcRoot str
 	}
 	defer imageDir.Close()
 
+	sysboxMounts := &[]string{
+		"/lib/modules/6.5.0-1017-aws",
```
Comment (nravic) [inline container/checkpoint.go:1315]: this isn't always true

File: api/dump.go (hunk 4 above)
Comment (bsmithai) [reply api/dump.go:161]: No, a few lines down is:

```
if pid != 0 {
	crPid = int(pid)
} else {
	crPid = runcContainer.Pid
}
```

File: container/checkpoint.go (hunk 5 above)
Comment (bsmithai) [reply container/checkpoint.go:1315]: Deleted this, this was an artifact

File: cedana-helper.Dockerfile (hunk 3 above)
Comment (nravic) [reply cedana-helper.Dockerfile:20]: this isn't fixed yet

Comment (bsmithai) [reply cedana-helper.Dockerfile]: fixed

**Review by nravic — APPROVED** [review]
banger

**Pull request merged**